Add phpstan to magento2 harness

andytson-inviqa opened the pull request.

andytson-inviqa committed:
Add phpstan to magento2 harness

kierenevans reviewed (dismissed):
Code changes are okay, but phar stream wrapper not happy in the builds - even for behat?!

andytson-inviqa committed:
Change magento2 harness to use non-shim phpstan

Resolves an issue with the need for restoring phar wrappers and is only suitable for Magento 2.3+
Magento 2.2.8 will need to continue to use the shim with patches to ignore the phpstan-shim autoload

kierenevans reviewed (approved):
Ace!
